lms/services/upsert.py

```diff
@@ -0,0 +1,94 @@
+"""A helper for upserting into DB tables."""
+
+from copy import deepcopy
+from typing import List
+
+from sqlalchemy import inspect
+from sqlalchemy.dialects.postgresql import insert
+from zope.sqlalchemy import mark_changed
+
+
+def bulk_upsert(  # pylint:disable=too-many-arguments
```

Comment:
No changes from the existing code here except the necessary to take the parameters from the function ones instead of the model config.